src/v/cluster/types.h

```diff
@@ -481,15 +487,28 @@ struct begin_tx_request
       : ntp(std::move(ntp))
       , pid(pid)
       , tx_seq(tx_seq)
-      , transaction_timeout_ms(transaction_timeout_ms) {}
+      , transaction_timeout_ms(transaction_timeout_ms)
+      , tm_partition(model::partition_id(0)) {}
+
```

Comment:
It's a minor thing, but sorta think that you should force the caller to pass `partition_id(0)` so that code using the old style is self-documenting explicitly rather than relying a heuristic about which constructor is used. wdyt?

Reply:
You are right.
It is very helpfull not to forget about some places where we used previous behaviour